controllers/machinedeletionremediation_controller.go

```diff
@@ -576,6 +588,36 @@ func (r *MachineDeletionRemediationReconciler) getMachineOwnerNodes(ctx context.
 	return machineOwnerNodes, nil
 }
 
+func (r *MachineDeletionRemediationReconciler) getMachineNameNsFromCR(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation) (machineName, machineNs string, err error) {
+	var node *v1.Node
+	node, err = r.getNodeFromCR(ctx, remediation)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			r.Log.Error(err, nodeNotFoundErrorMsg, "node name", remediation.Name)
+			err = nodeNotFoundError
+		}
+		return "", "", err
+	}
+
+	machineName, machineNs, err = getMachineNameNsFromNode(node)
+	if err != nil {
+		r.Log.Error(err, "could not get Machine Name NS from Node", "node", node.Name, "annotations", node.GetAnnotations())
+		return "", "", unrecoverableError
+	}
+	return machineName, machineNs, nil
+}
+
+func (r *MachineDeletionRemediationReconciler) getMachineNameNsFromOwnerReference(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation) (machineName, machineNs string) {
+	for _, owner := range remediation.GetOwnerReferences() {
+		if owner.Kind == machineKind {
```

Comment:
I think we are missing the edge case where MDR template is set at `openshift-machine-api`  - in which case NHC will create an ownerRefrence with a Machine, see [here](https://redhat-internal.slack.com/archives/C03M5GKJNBA/p1700595363777979?thread_ts=1700595331.969259&cid=C03M5GKJNBA) for more details.
In SNR we've verified this is not the case by making sure an ownerRef.Kind == "NodeHealthCheck"
does not exit - I think we can do the same here.

Reply:
I noted that in my review as well, but I think it doesn't make a difference here, because we don't look at the CR name if we find the Machine here in the OwnerRefs. It doesn't matter then if the CR is created for NHC or MHC.